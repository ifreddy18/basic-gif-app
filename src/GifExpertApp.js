import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const [categories, setCategories] = useState(['One Piece']);
    const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h2>GiftExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
			<hr />

            {/* <button onClick={ handleAdd }>Add category</button> */}

            <ol>
                {
                    // categories.map( category => <li key={ category }>{ category }</li>)
                    categories.map( category => (
                        <GifGrid 
                            key={ category }
                            category={ category }
                        />
                    ))
                }
            </ol>
		</>
	);
};

export default GifExpertApp;