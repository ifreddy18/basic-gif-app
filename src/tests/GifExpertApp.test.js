import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Pruebas en <GifExpertApp />', () => {

    test('Se debe mostrar el componene correctamente', () => {
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper ).toMatchSnapshot();
    })

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['Black Clover', 'One Piece'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />)
    
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    

})