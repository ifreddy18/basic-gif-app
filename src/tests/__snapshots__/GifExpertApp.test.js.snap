// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Black Clover"
      key="Black Clover"
    />
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Se debe mostrar el componene correctamente 1`] = `
<Fragment>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
